jobs/golangapiserver/spec

```diff
@@ -66,6 +66,13 @@ properties:
        - broker_username: user2
          broker_password: password2
     default: ''
+  autoscaler.apiserver.broker.credential_type:
+    description: |
+      The list of credential types to authenticate with custom metrics api. If credential_type "binding-secret" is used, then username and password are generated by the broker. Also, credential_type "binding-secret" is default, if no credential_type is provided.
+      example:
+       - credential_type: binding-secret
+       - credential_type: x509
+    default: 'binding-secret'
```

Comment:
What we configure here is the _default_ credential type. so I would name it this way.

```suggestion
  autoscaler.apiserver.broker.default_credential_type:
    description: |
      The default credential type generated to authenticate with the custom metrics API. if no credential type is explicitly set.
      Allowed values:
       - binding-secret
       - x509
      If credential-type "binding-secret" is set, then username and password are generated by the broker. 
      If credential-type "x509" is set, only instance identity credentials may be used.
    default: 'binding-secret'
```